proof/refine/RISCV64/ArchAcc_R.thy

```diff
@@ -498,14 +498,30 @@ lemma getPTE_wp:
   by (clarsimp simp: getObject_def split_def loadObject_default_def in_magnitude_check
                      in_monad valid_def obj_at'_def objBits_simps)
 
-lemma pt_at_lift:
-  "corres_inst_eq ptr ptr' \<Longrightarrow> \<forall>s s'. (s, s') \<in> state_relation \<longrightarrow> True \<longrightarrow>
-   (pspace_aligned s \<and> pspace_distinct s \<and> pt_at ptr s \<and> ptr = ptr') \<longrightarrow>
-    \<top> s' \<longrightarrow> page_table_at' ptr' s'"
-  by ( fastforce intro!: page_table_at_cross)
+(* only applies when ptr is available in abstract and concrete guard at the same time *)
+lemma pt_at_lift_eq:
+  "\<forall>s s'. (s, s') \<in> state_relation \<longrightarrow>
+          (pspace_aligned s \<and> pspace_distinct s \<and> pt_at ptr s) \<longrightarrow>
+          \<top> s' \<longrightarrow>
+          page_table_at' ptr s'"
```

Comment:
off-topic, but why is the format for these rules so strange? Especially for `pt_at_lift_relation` the combination of meta and object implication doesn't look obvious to me

Reply:
These are left-overs from Dan's very early attempts at cross-over rules. They are used only exactly here and nowhere else. In the long term we should actually remove them, I just wanted to make sure the new method could deal with the same kinds of goals that come out of these (and there were actually a few subtleties there, because when you cross over things, the guards might not depend on the correct green variables and so one).

Reply:
if so, then these ones could use a FIXME